# The Capstone Grading Curve (Goodness of Fit Test)
### Objective:

Analyze the final grade distributions of two separate Data Science cohorts (Class A and Class B). Compare their physical grade tallies against the university's strict historical blueprint (a standard bell curve) to determine which class followed expectations and which class completely broke the mold.

>Test Used: Chi-Square Goodness of Fit Test

>Performed by : Raghav Katyal on 30 September 2026

### The Historical Blueprint

The Data Science department dictates that in any given class of 50 students, the final grades follows this historical blueprint:
* 10% Excellent (5 students)
* 20% Good (10 students)
* 40% Average (20 students)
* 20% Poor (10 students)
* 10% Fail (5 students)

In [22]:
import numpy as np
from scipy import stats

In [23]:
expected_grades = [5,10,20,10,5]

#Class A Grades (Close to Blueprint)
classA_grades = [4,11,22,10,3]

chi_stat,p_value = stats.chisquare(f_obs=classA_grades,f_exp=expected_grades)
print(f"Chi-Square Score : {chi_stat:.2f}")
print(f"P-value : {p_value:.5f}")

Chi-Square Score : 1.30
P-value : 0.86138


>Observation : A Chi-Square score of 1.30 and p-value score of 0.86138 shows no strong evidence against the goodness of fit ; data passes the goodness of fit test. The Class A performed very close to the historical blueprint.

In [24]:
#Class B Grades (The Overachievers)
classB_grades = [20,15,7,7,1]

chi_stat,p_value = stats.chisquare(f_obs=classB_grades,f_exp=expected_grades)
print(f"Chi-Square Score : {chi_stat:.2f}")
print(f"P-value : {p_value:.5f}")

Chi-Square Score : 60.05
P-value : 0.00000


>Observation : A Chi-Square score of 60.06 and p-value score of 0.00 clearly shows strong evidence against the goodness of fit ; data fails the goodness of fit test. The Class B performed very different to the historical blueprint.